src/payments/search_builder.rs

```diff
@@ -0,0 +1,122 @@
+use async_stream::stream;
+use reqwest::Method;
+use std::pin::Pin;
+use tokio_stream::Stream;
+
+use super::{
+    get_builder::PaymentGetBuilder,
+    types::{PartialPaymentResult, PaymentResponse, PaymentSearchOptions, PaymentSearchResponse},
+};
+use crate::{
+    client::MercadoPagoClient,
+    common::{resolve_json, MercadoPagoRequestError},
+};
+
+/// Builder for search payments
+///
+/// # Arguments
+///
+/// * `options` - Options to search for payments
+///
+/// # Example
+/// ```
+/// PaymentSearchBuilder(
+///     PaymentSearchOptions {
+///         limit: Some(10),
+///         sort: Some(PaymentSearchSort::DateLastUpdated)
+///         ..Default::default()
+///     }
+/// )
+/// ```
+///
+/// # Docs
+/// <https://www.mercadopago.com.br/developers/pt/reference/payments/_payments_search/get>
+pub struct PaymentSearchBuilder(pub PaymentSearchOptions);
+
+impl PaymentSearchBuilder {
+    /// Send the request
+    pub async fn send<'a>(
```

Comment:
E também renomeia essa função para `fetch_all_streamed`

```suggestion
    pub async fn fetch_all_streamed<'a>(
```

Reply:
Arrumado no commit [e28f3b1](https://github.com/vipago/mpago/pull/2/commits/e28f3b1ac668dc1aa5a7d7953cc4cc90047f51be)